src/Raven.Server/Documents/TcpHandlers/SubscriptionConnection.cs

```diff
@@ -380,10 +380,19 @@ public async Task Run(TcpConnectionOptions tcpConnectionOptions, IDisposable sub
                         {
                             await InitAsync();
                             _subscriptionConnectionsState = await TcpConnection.DocumentDatabase.SubscriptionStorage.OpenSubscriptionAsync(this);
-                            (disposeOnDisconnect, registerConnectionDurationInTicks) = await SubscribeAsync();
+                            await _subscriptionConnectionsState.TakeConcurrentConnectionLock(this);
+                            try
+                            {
+                                (disposeOnDisconnect, registerConnectionDurationInTicks) = await SubscribeAsync();
```

Comment:
inside SubscribeAsync you are adding pending connections, now this will be always 1?

Reply:
Dont you need to do it as a part of taking the lock?Just before it happens?

Reply:
fixed